restful-tango/tangoREST.py

```diff
@@ -301,12 +302,11 @@ def upload(self, key, courselab, file, body):
             labPath = self.getDirPath(key, courselab)
             try:
                 if os.path.exists(labPath):
+                    fileMD5 = hashlib.md5(body).hexdigest()
+                    filesInDir = self.computeMD5(labPath)
+                    if file in filesInDir and filesInDir[file] == fileMD5:
+                        return self.status.file_exists
```

Comment:
Also, this is only for one call to upload (i.e. one file upload), it doesn't stop the whole upload process for a job at all because the client can call upload many times.